expression/builtin_string.go

```diff
@@ -1345,56 +1347,198 @@ func trimRight(str, remstr string) string {
 	}
 }
 
+type lpadFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *lpadFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpString, tpInt, tpString)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if types.IsBinaryStr(args[0].GetType()) || types.IsBinaryStr(args[2].GetType()) {
+		types.SetBinChsClnFlag(bf.tp)
+		sig := &builtinLpadBinarySig{baseStringBuiltinFunc{bf}}
+		return sig.setSelf(sig), nil
```

Comment:
should we check `mysql.HasBinaryFlag(args[i].GetType().Flag)` here,
and set bf.tp.Flag |= mysql.BinaryFlag if it's true